### PR TITLE
[FlagGems Operator Development Competition] Added leaky relu operator

### PR DESCRIPTION
**Pull request by YoihenBachu**
### PR Category

- [x] Operator

### Type of Change

- [x] New Feature

### Description
Implements LeakyReLU (Leaky Rectified Linear Unit) operator for FlagGems using Triton.
This PR adds support for the LeakyReLU activation function with:

- Forward operation: torch.nn.functional.leaky_relu(input, negative_slope=0.01)
- In-place operation: torch.nn.functional.leaky_relu(input, negative_slope=0.01, inplace=True)
- Support for FLOAT_DTYPES (float16, bfloat16, float32, float64)
- Configurable negative slope parameter (default: 0.01)
- Full autograd integration

The implementation follows the FlagGems operator development pattern with efficient Triton kernels for both forward and backward passes.

### Issue

- Associated with FlagGems Operator Development Competition
- Implements LeakyReLU operator as part of activation function operators

### Progress

- [] Change is properly reviewed (1 reviewer required, 2 recommended).
- [x] Change is responded to an issue.
- [x] Change is fully covered by a UT.

### Performance
Performance benchmarks show competitive results against PyTorch native implementation:

- Tested across multiple tensor shapes: 2D (1024×1 to 1024×65536) and 3D (64×64×1 to 64×64×4096)
- Tested across all float dtypes: float16, bfloat16, float32, float64
- Speedup ≥0.9x compared to PyTorch
- Forward, in-place, and backward variants tested

**Test Coverage:**

- Basic functionality: Multiple shapes (0D-5D), all float dtypes
- Negative slope variants: Default (0.01), zero, small (0.001), standard (0.2), large (0.5)
- Edge cases: Extreme aspect ratios, very small/large matrices, scalar tensors
- Memory layouts: Contiguous, transposed, strided, permuted
- Special values: inf, -inf, nan, zeros, positive/negative inputs
- Operator variants: Out-of-place, in-place
- Backward pass: Gradient computation with various negative slopes

Run benchmark with:
```bash
pytest benchmark/test_unary_pointwise_perf.py -m leaky_relu -s
```

leaky_relu_bench1
leaky_relu_bench2

Run tests with:
```bash
pytest tests/test_unary_pointwise_ops.py -m leaky_relu
```

leaky_relu_tests

### CODE REVIEW

**Comment by douxetpur** [comment]
Hi, thank you for your participation in the FlagOS Open Computing Competition 🙏

After review, we have chosen to move forward with a different implementation for this operator. As a result, this PR will be closed.

We appreciate your time and contribution, and hope to see more of your submissions in the future.
